pytups/tuplist.py

```diff
@@ -100,14 +100,33 @@ def to_dict(
         """
         from . import superdict as sd
 
-        if result_col is None:
+        if len(self) == 0:
+            return sd.SuperDict()
+        # Handle case of list of dict
+        if isinstance(self[0], dict):
+            if indices is None:
+                raise ValueError(
+                    "For a list of dict, to_dict require indices to be specified"
+                )
+            if result_col is None:
+                if type(indices) is not list or len(indices) == 1:
+                    return sd.SuperDict({d[indices]: d for d in self})
+                else:
+                    return sd.SuperDict({tuple(d[i] for i in indices): d for d in self})
```

Comment:
i do not understand why this logic is only valid for lists of dictionaries. It should also be valid for tuplists.